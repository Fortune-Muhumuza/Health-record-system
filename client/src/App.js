import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import Navbar from './components/navigation/navbar';
import login from './components/authComponents/login';
import register from './components/authComponents/register';
import home from './components/homePage'
import './App.css';

function App() {
  return (
    <Router>
      <div className="container">
     <Navbar />
      <br />
      <Route path='/login' exact component={login} />
      <Route path='/register' exact component={register} />
      <Route path='/homePage' exact component={home} />
      </div>
    </Router>
  );
}

export default App;
