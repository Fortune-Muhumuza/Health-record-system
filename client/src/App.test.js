import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import App from './App';

test('renders learn react', () => {
  const { getByText } = render(<App />);
  const Element = getByText(/learn react/i);
  expect(Element).toBeInTheDocument();
});

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
});

afterEach(cleanup);

it('increments counter', () => {
  const { getByTestId } = render(<App />);

  fireEvent.click(getByTestId('button-up'));

  expect(getByTestId('counter')).toHaveTextContent('2');
});

it('decrements counter', () => {
  const { getByTestId } = render(<App />);

  fireEvent.click(getByTestId('button-down'));

  expect(getByTestId('counter')).toHaveTextContent('-1');
});


afterEach(cleanup);

it('adds yess to the array', () => {
  const { getByTestId } = render(<App />);

  fireEvent.click(getByTestId('addToArray'));

  expect(getByTestId('array')).toHaveTextContent('ab');
});
