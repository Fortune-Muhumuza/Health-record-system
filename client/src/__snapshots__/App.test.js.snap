// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="makeStyles-root-4"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <h6
            class="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
          >
            HMS
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                href="/login"
                style="text-decoration: none; color: white;"
              >
                Sign in
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                href="/register"
                style="text-decoration: none; color: white;"
              >
                Register
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                href="/homePage"
                style="text-decoration: none; color: white;"
              >
                Home
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <a
                href="/receptionPage"
                style="text-decoration: none; color: white;"
              >
                Reception
              </a>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
    <br />
  </div>
</DocumentFragment>
`;
